### PR TITLE
reverting to 05208af

### PR DESCRIPTION
**Pull request by DZarudnyi**

### CODE REVIEW

File: src/main/java/com/example/books/dto/CreateBookRequestDto.java [hunk 1]
```diff
@@ -1,13 +1,20 @@
...
 public class CreateBookRequestDto {
+    @NotNull
     private String title;
+    @NotNull
     private String author;
+    @NotNull
     private String isbn;
+    @NotNull
+    @Min(0)
     private BigDecimal price;
```
Comment (Sanderfuck) [inline src/main/java/com/example/books/dto/CreateBookRequestDto.java:18]: Poor validation make it more custom

File: src/main/java/com/example/books/dto/CreateBookRequestDto.java [hunk 2]
```diff
@@ -1,13 +1,20 @@
...
     private String description;
```
Comment (Sanderfuck) [inline src/main/java/com/example/books/dto/CreateBookRequestDto.java:19]: Validate description

File: src/main/java/com/example/books/dto/UpdateBookRequestDto.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+@Data
+public class UpdateBookRequestDto {
+    private String title;
+    private String author;
+    private String isbn;
+    @Min(0)
+    private BigDecimal price;
+    private String description;
+    private String coverImage;
```
Comment (Sanderfuck) [inline src/main/java/com/example/books/dto/UpdateBookRequestDto.java:15]: Is it validation?

File: src/main/java/com/example/books/model/Book.java [hunk 4]
```diff
@@ -14,31 +14,24 @@
...
     private String author;
-
     @Column(unique = true)
     @NotNull
     private String isbn;
-
     @NotNull
     private BigDecimal price;
-
     private String description;
```
Comment (Sanderfuck) [inline src/main/java/com/example/books/model/Book.java:33]: What about description?

**Review by Sanderfuck — CHANGES_REQUESTED** [review]

File: src/main/java/com/example/books/dto/CreateBookRequestDto.java (hunk 2 above)
Comment (DZarudnyi) [reply src/main/java/com/example/books/dto/CreateBookRequestDto.java:19]: image
What exactly are you expecting to validate? A string field without constraints that can be null?

File: src/main/java/com/example/books/dto/UpdateBookRequestDto.java (hunk 3 above)
Comment (DZarudnyi) [reply src/main/java/com/example/books/dto/UpdateBookRequestDto.java:15]: Have you checked the name of the DTO? It is for an update request, all but one field can be null.

File: src/main/java/com/example/books/model/Book.java (hunk 4 above)
Comment (DZarudnyi) [reply src/main/java/com/example/books/model/Book.java:33]: Check my comment higher

File: src/main/java/com/example/books/dto/CreateBookRequestDto.java (hunk 1 above)
Comment (DZarudnyi) [reply src/main/java/com/example/books/dto/CreateBookRequestDto.java:18]: And how exactly it should look like? Maybe you should add some description of the fields, so I would know what constraints it should have?

File: .mvn/wrapper/maven-wrapper.jar
Comment (olekskov) [inline .mvn/wrapper/maven-wrapper.jar:1]: .mvn folder should be excluded from repository
Please make relevant .gitignore changes and remove from PR

File: mvnw
Comment (olekskov) [inline mvnw:1]: same, this shouldn't be part of PR and should be excluded from git

File: mvnw.cmd
Comment (olekskov) [inline mvnw.cmd:1]: same

File: src/main/java/com/example/books/model/Book.java [hunk 5]
```diff
@@ -14,31 +14,24 @@
...
-@Where(clause = "is_deleted=false")
+@SQLDelete(sql = "UPDATE books SET is_deleted = TRUE WHERE id = ?")
+@Where(clause = "is_deleted=FALSE")
 @Table(name = "books")
 public class Book {
     @Id
     @GeneratedValue(strategy = GenerationType.IDENTITY)
     private Long id;
-
     @NotNull
```
Comment (olekskov) [inline src/main/java/com/example/books/model/Book.java:24]: Validation constraints should validate incoming information so used in DTOs
In models we don't use them

File: src/main/java/com/example/books/dto/CreateBookRequestDto.java (hunk 1 above)
Comment (olekskov) [reply src/main/java/com/example/books/dto/CreateBookRequestDto.java:18]: There is no any prescribed approach. We use either technical requirements or common sense.
Most used predefined annotations are '@NotNull', '@NotEmpty', '@NotBlank', '@Length'.
Here '@Isbn' would also be very natural

**Review by olekskov — CHANGES_REQUESTED** [review]

**Review by olekskov — APPROVED** [review]

**Pull request merged**